ext/winevt/winevt_utils.cpp

```diff
@@ -787,7 +836,14 @@ render_system_event(EVT_HANDLE hEvent, BOOL preserve_qualifiers)
 
   if (EvtVarTypeNull != pRenderedValues[EvtSystemUserID].Type) {
     if (ConvertSidToStringSid(pRenderedValues[EvtSystemUserID].SidVal, &pwsSid)) {
-      rbstr = rb_utf8_str_new_cstr(pwsSid);
+      CHAR *expandSID;
+      if (expandSID_p &&
+          ExpandSIDWString(pRenderedValues[EvtSystemUserID].SidVal,
+                           &expandSID) == 0) {
+        rbstr = rb_utf8_str_new_cstr(expandSID);
+      } else {
+        rbstr = rb_utf8_str_new_cstr(pwsSid);
+      }
       rb_hash_aset(hash, rb_str_new2("UserID"), rbstr);
```

Comment:
Agreed.

Reply:
I don't think so. This feature is intended to processor for SID on elastic beat. Preserving the original data is not fulfilled the requirements. Or, Providing preserving the original SID would be better?

Reply:
I added a commit which preserves the original SID if preserveSID option is enabled. Default is off. This is because it assumes for the high throughput environment. So, just adding the new parameter increases the amount of logs. It's sick for our usecase.
https://github.com/fluent-plugins-nursery/winevt_c/pull/46/commits/a0b3608f42836ce674db1dc05aef9f69df3b2cbd

Reply:
Thanks for considering it! I see.

> Default is off. This is because it assumes for the high throughput environment.

I understand your points, but I have concerns in terms of keeping compatibility.
If we prioritize compatibility, it would be better to make the default on.

Is there a problem with the default being on and setting it to off to improve throughput?

Reply:
It seems there is no issues for now. If we use this option, we _must_ write down the document _how affects the collecting hash structure_.

Reply:
... In fluent-plugin-windows-eventlog2.

Reply:
Defaulting on here: https://github.com/fluent-plugins-nursery/winevt_c/pull/46/commits/eef4e49b628201e50a5bdbfe7a4b445697c97a52

Reply:
Thanks!